runtime/jvmti/jvmtiThread.cpp

```diff
@@ -1382,27 +1401,32 @@ jvmtiSuspendResumeCallBack(J9VMThread *vmThread, J9MM_IterateObjectDescriptor *o
 			J9JavaVM *vm = vmThread->javaVM;
 			J9VMThread *targetThread = NULL;
 			j9object_t carrierThread = (j9object_t)J9VMJAVALANGVIRTUALTHREAD_CARRIERTHREAD(vmThread, vthread);
+			U_64 isSuspendedFlag = J9OBJECT_U64_LOAD(vmThread, vthread, vm->internalSuspendStateOffset);
+			BOOLEAN inTransition = J9_ARE_ANY_BITS_SET(isSuspendedFlag, ~0x1);
 			if (NULL != carrierThread) {
 				targetThread = J9VMJAVALANGTHREAD_THREADREF(vmThread, carrierThread);
 				Assert_JVMTI_notNull(targetThread);
+			} else if (inTransition) {
+				/* Virtual thread during transition. */
+				targetThread = (J9VMThread *)(isSuspendedFlag & ~0x1);
 			}
 			if (data->is_suspend) {
-				if ((NULL != targetThread) && (vthread == targetThread->threadObject)) {
+				J9OBJECT_U64_STORE(vmThread, vthread, vm->internalSuspendStateOffset, (isSuspendedFlag | 1));
```

Comment:
we should use the new macro here.
```suggestion
				J9OBJECT_U64_STORE(vmThread, vthread, vm->internalSuspendStateOffset, (isSuspendedFlag | J9_VIRTUALTHREAD_INTERNAL_STATE_SUSPENDED));
```